### PR TITLE
Add the classical approach using linear programming

### LINKED ISSUE #1: Add classical approach
Add the created classical approach to the Jupyter notebook to solve the planning problem.

### PR DESCRIPTION
**Pull request by Iodine98**
This PR includes the classical approach to the scheduling problem by defining the constraints and adding these to a linearly defined problem using [pulp](https://coin-or.github.io/pulp/).

### CODE REVIEW

**Comment by Iodine98** [comment]
Please check carefully if the notebook still runs as expected. I did not check if the backends of the quantum computers run, because I do not have an account at Amazon, IBM, etc.

**Comment by Iodine98** [comment]
The linting check failed on a package that I did not introduce:

image
****

**Comment by costastf** [comment]
There are two depenencies jupyter and pexpect that are added to pipenv but not used.

**Comment by costastf** [comment]
> Please check carefully if the notebook still runs as expected. I did not check if the backends of the quantum computers run, because I do not have an account at Amazon, IBM, etc.

The backends are irrelevant of the usage of probabilities. If the code works up until the export of the bound_c then the probabilities work fine.

File: planning.ipynb
Comment (huisman) [inline planning.ipynb:223]: The step of selecting/creating the correct kernel is indeed missing from the description.
Admittedly, I didn´t use VSCode so maybe the Jupyter extension already takes care of this, but I had to create and select a kernel for the pipenv environment I used.

**Comment by huisman** [comment]
> The backends are irrelevant of the usage of probabilities. If the code works up until the export of the bound_c then the probabilities work fine.

This PR does not touch the probabilities as far as I can see?

**Comment by costastf** [comment]
> > The backends are irrelevant of the usage of probabilities. If the code works up until the export of the bound_c then the probabilities work fine.
> 
> This PR does not touch the probabilities as far as I can see?

No it does not, i was just doing a hundred different things at the same time and was thinking of a comment for the previous PR 😊

**Comment by Iodine98** [comment]
The fixing branch has been merged in the current one. Everything runs after merge. Can @costastf and/or @huisman verify that it works; by approving the review? If it has been approved, I will merge it.

**Review by costastf — APPROVED** [review]

**Comment by costastf** [comment]
Wait, why do we have Jupiter on the main dependencies?

**Comment by Iodine98** [comment]
> Wait, why do we have Jupyter on the main dependencies?

Because I tried executing the notebook on a Jetbrains IDE (Pycharm) and it asked me to install Jupyter to run the notebook. Therefore, I think Jupyter should be added as a (dev-)(dependency.

**Comment by costastf** [comment]
I would think that that should be installed as an addon on pycharm itself and not be brought in as a dependency of the project.

**Comment by costastf** [comment]
The exact same approach that has been taken for vscode. The Jupiter addon is installed on the editor and not part of the project.

**Comment by Iodine98** [comment]
> The exact same approach that has been taken for vscode. The Jupiter addon is installed on the editor and not part of the project.

Unfortunately, that is not how Jetbrains implemented Jupyter notebook support in Pycharm. They have support [built-in](https://www.jetbrains.com/help/pycharm/jupyter-notebook-support.html#ui) given that the `Jupyter` package has been installed in the current Python environment.

**Comment by costastf** [comment]
> > The exact same approach that has been taken for vscode. The Jupiter addon is installed on the editor and not part of the project.
> 
> Unfortunately, that is not how Jetbrains implemented Jupyter notebook support in Pycharm. They have support [built-in](https://www.jetbrains.com/help/pycharm/jupyter-notebook-support.html#ui) given that the `Jupyter` package has been installed in the current Python environment.

On  my Pycharm i do not have jupyter installed as a dependency of the project while having this addon installed on the editor 
Screenshot 2023-12-18 at 14 38 54
and everything works as expected. Could you please validate that this addon in enabled on your IDE?

File: planning.ipynb
Comment (Iodine98) [reply planning.ipynb:223]: I have added that instruction in the latest commit.

**Comment by Iodine98** [comment]
@costastf, there's something with the `pexpect` dependency that I cannot seem to solve. On my local machine, it does not ask for this module. Can you have a look at my latest commit?

**Comment by costastf** [comment]
> @costastf, there's something with the `pexpect` dependency that I cannot seem to solve. On my local machine, it does not ask for this module. Can you have a look at my latest commit?

on it

**Pull request merged**
